Mathlib/RingTheory/Localization/Defs.lean

```diff
@@ -719,6 +718,28 @@ theorem map_injective_of_injective (h : Function.Injective g) [IsLocalization (M
     Function.Injective (map Q g M.le_comap_map : S → Q) :=
   (toLocalizationMap M S).map_injective_of_injective h (toLocalizationMap (M.map g) Q)
 
+/--
+Another version of `IsLocalization.map_injective_of_injective` that requires that there is no zero
+divisors but is more general for the choice of the localization submodule.
+-/
+theorem map_injective_of_injective' {A : Type*} [CommRing A] {B : Type*} [CommRing B] {f : A →+* B}
```

Comment:
I'm not sure if you should be adding `theorem`s to a file called `Defs`. Especially when it increases imports. Does changing to `FaithfulSMul` solve the import problem?

Reply:
It turns out it can go to `RingTheory.Localization.Basic` without any additional imports. Strange that `find_home` didn't see that..

Reply:
I usually use `#min_imports in` instead of find home.